charts/redpanda/client_test.go

```diff
@@ -45,3 +46,494 @@ func (c *Client) ClusterConfig(ctx context.Context) (redpanda.ClusterConfig, err
 
 	return config, nil
 }
+
+func (c *Client) CreateTopic(ctx context.Context, topicName string) error {
+	pod, err := c.getStsPod(ctx, 0)
+	if err != nil {
+		return err
+	}
+
+	var out bytes.Buffer
+	if err = c.Ctl.Exec(ctx, pod, kube.ExecOptions{
+		Command: []string{"bash", "-c", fmt.Sprintf(`rpk topic create %s -r 3 -p 3`, topicName)},
+		Stderr:  &out,
+	}); err != nil {
+		return err
+	}
+
+	var config map[string]any
+	if err = yaml.Unmarshal(out.Bytes(), &config); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) KafkaProduce(ctx context.Context, input, topicName string) (string, error) {
+	pod, err := c.getStsPod(ctx, 0)
+	if err != nil {
+		return "", err
+	}
+
+	var out, eb bytes.Buffer
+	if err = c.Ctl.Exec(ctx, pod, kube.ExecOptions{
+		Command: []string{"bash", "-c", fmt.Sprintf(`echo %s | rpk topic produce %s`, input, topicName)},
+		Stdout:  &out,
+		Stderr:  &eb,
+	}); err != nil {
+		return "", err
+	}
+
+	if eb.String() != "" {
+		return "", fmt.Errorf(eb.String())
+	}
+
+	return out.String(), nil
+}
+
+func (c *Client) KafkaConsume(ctx context.Context, topicName string) (map[string]any, error) {
+	pod, err := c.getStsPod(ctx, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	var out, eb bytes.Buffer
+	if err = c.Ctl.Exec(ctx, pod, kube.ExecOptions{
+		Command: []string{"bash", "-c", fmt.Sprintf(`rpk topic consume %s -n 1`, topicName)},
+		Stdout:  &out,
+		Stderr:  &eb,
+	}); err != nil {
+		return nil, err
+	}
+
+	if eb.String() != "" {
+		return nil, fmt.Errorf(eb.String())
+	}
+
+	var event map[string]any
+	if err = json.Unmarshal(out.Bytes(), &event); err != nil {
+		return nil, err
+	}
+
+	return event, nil
+}
+
+func (c *Client) GetClusterHealth(ctx context.Context, dot *helmette.Dot) (map[string]any, error) {
```

Comment:
I don't love the idea of using `*helmette.Dot` here. If there's a bug in some of the Values functions that would surface here and potentially obscure issues from our tests, specifically in the case of TLS configurations.

I think it would be a more robust testing strategy to instead have the Client accept the TLS certificates (extracted from the cluster if using the cert-manger integration) and then use those to actually dial into the various listeners of the Pod, piggy backing on `kubectl port-forward` to actually make the connection. Then we can rely on client libraries instead of having to template out CLI commands.

For an initial implementation thought, let's roll with it!

Reply:
I will add certificate to the Client.